### PR TITLE
[AIRFLOW-1077] Warn about subdag deadlock case

### PR DESCRIPTION
**Pull request by akosel**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-1077] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1077

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

Addresses a case that may be counter-intuitive to users where a subdag count greater than the concurrency limit can lead to deadlock. It also suggests a workaround for dealing with this issue.

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

It's a comment so there isn't anything to test.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

File: docs/concepts.rst [hunk 1]
```diff
@@ -554,6 +554,13 @@ Some other tips when using SubDAGs:
    effectively limit its parallelism to one. Using LocalExecutor can be
    problematic as it may over-subscribe your worker, running multiple tasks in
    a single slot
+-  do not create more SUBDags then your concurrency limit or the scheduler
```
Comment (saguziel) [inline docs/concepts.rst]: The capitalization is all over the place in this doc, let's stick with "SubDag" since that is what is in code.

**Comment by saguziel** [comment]
I think you should create a new issue for this rather than use that one

**Comment by tispratik** [comment]
I think i am running into the same issue. Airflow says the subdags are running, but none of the steps within it are executing. When i restarted airflow, some progress happened, but then it got blocked again.

**Comment by SamWildmo** [comment]
I think i also encountered this but it happened once so i cant be sure.

**Comment by yegeniy** [comment]
This is a great warning for those who are not on the latest version, but the default executor for SubDags is now the SequentialExecutor.

**Comment by SamWildmo** [comment]
@yegeiny if someone can overwrite it to local executor then this PR is still needed. If not then maybe commiters should close this.

**Comment by yegeniy** [comment]
Great point. I think it's still totally over-writeable.

**Pull request closed without merging**